### PR TITLE
feat: cardinal pretty logging, improve goroutine reliability, kill --watch

### PR DESCRIPTION
**Pull request by smsunarto**
### What changed?

- `world cardinal dev` now use pretty logging on latest Cardinal version. This is done by setting `CARDINAL_PRETTY_LOG` to true.
- Refactors to goroutine management in `world cardinal start` and `world cardinal dev` to make it more reliable using error group.
- Disable printing the usage guide when a command returns an error.
- Various string reformatting using lipgloss to make things look pretty.
- Prevent exit code 130, 137, 143 from being treated as an error since it's an expected exit code from Docker when it receives termination signal (i.e. Ctrl + C, SIGKILL, SIGTERM, etc)
- Kill `world cardinal dev --watch` flag because runner is ugly and buggy af

### How to test?

1. Run `world cardinal dev` and `world cardinal start` and try to break it. This includes running it against various edge cases (i.e. Redis already running, kill redis when cardinal is running, etc)

### CODE REVIEW

**Comment by smsunarto** [comment]
* **#64** <a href="https://app.graphite.dev/github/pr/Argus-Labs/world-cli/64?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @smsunarto and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by zulkhair** [comment]
I'm testing to run with `world cardinal dev --watch`. It got stuck when i'm tryin to send sigint.
Screenshot from 2024-05-17 11-23-27

File: cmd/world/cardinal/dev.go [hunk 1]
```diff
@@ -182,165 +116,182 @@ var devCmd = &cobra.Command{
 /////////////////
 
 func init() {
-	devCmd.Flags().Bool(flagWatch, false, "Dev mode with hot reload support")
-	devCmd.Flags().Bool(flagNoEditor, false, "Run Cardinal without the editor")
+	devCmd.Flags().Bool(flagNoEditor, false, "Run Cardinal without the editor (default: false)")
+	devCmd.Flags().Bool(flagPrettyLog, true, "Run Cardinal with pretty logging (default: true)")
```
Comment (zulkhair) [inline cmd/world/cardinal/dev.go:120]:
```suggestion
	devCmd.Flags().Bool(flagNoEditor, false, "Run Cardinal without the editor (default false)")
	devCmd.Flags().Bool(flagPrettyLog, true, "Run Cardinal with pretty logging")
```

default true printed twice on `--pretty-log` flag

**Review by zulkhair — APPROVED** [review]
Tested all the functionality. LGTM

File: cmd/world/cardinal/start.go [hunk 2]
```diff
@@ -100,42 +103,41 @@ This will start the following Docker services and its dependencies:
...
+		// Start the World Engine stack
+		group.Go(func() error {
+			if err := teacmd.DockerStartAll(cfg); err != nil {
+				return eris.Wrap(err, "Encountered an error with Docker")
+			}
+			return eris.Wrap(ErrGracefulExit, "Stack terminated")
+		})
+
+		// Start Cardinal Editor is flag is set to true
+		if runEditor {
```
Comment (zulkhair) [inline cmd/world/cardinal/start.go:125]: another small nit.
on `world cardinal start` there is no information about which port that cardinal editor used, it will be better if the information is printed like on `world cardinal dev`.

Comment (smsunarto) [reply cmd/world/cardinal/start.go:125]: Let's do that in a separate PR

Comment (zulkhair) [reply cmd/world/cardinal/start.go:125]: ok
